samples/net/sockets/echo_client/overlay-ot.conf

```diff
@@ -30,24 +28,9 @@ CONFIG_OPENTHREAD_CHANNEL=26
 CONFIG_NET_CONFIG_MY_IPV6_ADDR="fdde:ad00:beef::1"
 CONFIG_NET_CONFIG_PEER_IPV6_ADDR="fdde:ad00:beef::2"
 
-# Thread by default registers quite a lot addresses.
-CONFIG_NET_IF_UNICAST_IPV6_ADDR_COUNT=6
-CONFIG_NET_IF_MCAST_IPV6_ADDR_COUNT=8
-
-# Other OpenThread dependencies
-CONFIG_FLASH=y
-CONFIG_FLASH_PAGE_LAYOUT=y
-CONFIG_FLASH_MAP=y
-CONFIG_MPU_ALLOW_FLASH_WRITE=y
-CONFIG_NVS=y
-CONFIG_SETTINGS=y
-
-CONFIG_REBOOT=y
-
 # mbedTLS tweaks
 CONFIG_MBEDTLS_SSL_MAX_CONTENT_LEN=768
 
 # A sample configuration to enable Thread Joiner, uncomment if needed
-#CONFIG_OPENTHREAD_JOINER=y
+CONFIG_OPENTHREAD_JOINER=y
```

Comment:
I'd not enable joiner by default, the samples will not work out-of-the-box in that case.